pkg/oci/utils.go

```diff
@@ -161,61 +197,94 @@ func networkConfig(ocispec CompatOCISpec) (vc.NetworkConfig, error) {
 	return netConf, nil
 }
 
-// PodConfig converts an OCI compatible runtime configuration file
-// to a virtcontainers pod configuration structure.
-func PodConfig(runtime RuntimeConfig, bundlePath, cid, console string) (*vc.PodConfig, *CompatOCISpec, error) {
-	configPath := filepath.Join(bundlePath, "config.json")
+// getConfigPath returns the full config path from the bundle
+// path provided.
+func getConfigPath(bundlePath string) string {
+	return filepath.Join(bundlePath, "config.json")
+}
+
+// ParseConfigJSON unmarshals the config.json file.
+func ParseConfigJSON(bundlePath string) (CompatOCISpec, error) {
+	configPath := getConfigPath(bundlePath)
 	ociLog.Debugf("converting %s", configPath)
 
 	configByte, err := ioutil.ReadFile(configPath)
 	if err != nil {
-		return nil, nil, err
+		return CompatOCISpec{}, err
 	}
 
 	var ocispec CompatOCISpec
-	if err = json.Unmarshal(configByte, &ocispec); err != nil {
-		return nil, nil, err
+	if err := json.Unmarshal(configByte, &ocispec); err != nil {
+		return CompatOCISpec{}, err
 	}
 
-	rootfs := ocispec.Root.Path
-	if !filepath.IsAbs(rootfs) {
-		rootfs = filepath.Join(bundlePath, ocispec.Root.Path)
+	return ocispec, nil
+}
+
+// GetContainerType determines which type of container matches the annotations
+// table provided.
+func GetContainerType(annotations map[string]string) (vc.ContainerType, error) {
+	if containerType, ok := annotations[ContainerTypeKey]; ok {
+		return vc.ContainerType(containerType), nil
 	}
-	ociLog.Debugf("container rootfs: %s", rootfs)
 
-	cmd := vc.Cmd{
-		Args:         ocispec.Process.Args,
-		Envs:         cmdEnvs(ocispec, []vc.EnvVar{}),
-		WorkDir:      ocispec.Process.Cwd,
-		User:         strconv.FormatUint(uint64(ocispec.Process.User.UID), 10),
-		PrimaryGroup: strconv.FormatUint(uint64(ocispec.Process.User.GID), 10),
-		Interactive:  ocispec.Process.Terminal,
-		Console:      console,
+	ociLog.Errorf("Annotations[%s] not found, cannot determine the container type",
+		ContainerTypeKey)
+	return vc.UnknownContainerType, fmt.Errorf("Could not find container type")
+}
+
+// ContainerType returns the type of container and if the container type was
+// found from CRI servers annotations.
+func (spec *CompatOCISpec) ContainerType() (vc.ContainerType, bool, error) {
```

Comment:
Well now I think this is not needed anymore because according to the recent changes into CRI-O getting the shim properly terminated (and not remaining as a zombie process), we don't need to make a difference between those two cases. And it is in my opinion better if we can avoid to make a difference between a CRI-O and a Docker pod sandbox.
I am removing this `bool` return.